apps/web/components/eventtype/EventLimitsTab.tsx

```diff
@@ -109,9 +109,99 @@ const MinimumBookingNoticeInput = React.forwardRef<
   );
 });
 
-export const EventLimitsTab = ({ eventType }: Pick<EventTypeSetupProps, "eventType">) => {
-  const { t, i18n } = useLocale();
-  const formMethods = useFormContext<FormValues>();
+type ShouldLockDisableProps = {
+  disabled?: boolean;
+  LockedIcon?: React.ReactNode;
+  isLocked?: boolean;
+};
+
+type BookingLimitsProps = {
+  bookingLimitsLocked?: ShouldLockDisableProps;
+  sectionDescription?: string;
+  settingsToggleClass?: string;
+  onCheckedChange: (field: string, value: IntervalLimit, shouldDirty: boolean) => void;
+  childrenContainerClassName?: string;
+  children?: React.ReactNode;
+};
+export const BookingLimits = ({
+  bookingLimitsLocked = {},
+  sectionDescription = "limit_booking_frequency_description",
+  settingsToggleClass = "",
+  onCheckedChange,
+  childrenContainerClassName = "",
+  children,
+}: BookingLimitsProps) => {
+  const { t } = useLocale();
+  return (
+    <Controller
+      name="bookingLimits"
+      render={({ field: { value } }) => {
+        const isChecked = Object.keys(value ?? {}).length > 0;
+        return (
+          <SettingsToggle
+            toggleSwitchAtTheEnd={true}
+            labelClassName="text-sm"
+            title={t("limit_booking_frequency")}
+            {...bookingLimitsLocked}
+            description={t(sectionDescription)}
+            checked={isChecked}
+            onCheckedChange={(active) => {
+              if (active) {
+                onCheckedChange(
+                  "bookingLimits",
+                  {
+                    PER_DAY: 1,
+                  },
+                  true
+                );
+              } else {
+                onCheckedChange("bookingLimits", {}, true);
+              }
+            }}
+            switchContainerClassName={classNames(
+              "border-subtle mt-6 rounded-lg border py-6 px-4 sm:px-6",
+              isChecked && "rounded-b-none",
+              settingsToggleClass
+            )}
+            childrenClassName="lg:ml-0">
+            <div
+              className={classNames(
+                "border-subtle rounded-b-lg border border-t-0 p-6",
+                childrenContainerClassName
+              )}>
+              <IntervalLimitsManager
+                disabled={!!bookingLimitsLocked?.disabled}
+                propertyName="bookingLimits"
+                defaultLimit={1}
+                step={1}
+              />
+              {children}
+            </div>
+          </SettingsToggle>
+        );
+      }}
+    />
+  );
+};
+
+type FutureBookingLimitsProps = {
+  formMethods: UseFormReturn<any>;
```

Comment:
Ideally we shouldn't be using `any` but this component is being used in 2 different places, both having different type shape but with common fields used in this component, not able to work it out because `react-hook-form` throws error in watch and others